### PR TITLE
feat: Add charge confirmations

### PR DESCRIPTION
**Pull request by bemwen**
## Description

The purpose of this PR is to add charge confirmations to the message hub flow and remove old confirmation post office implementation.

* Adds available charge confirmation factory
* Adds charge confirmation specification
* Adds needed dependency injection
* Renames confirmer endpoint to be inline with new naming
* Removes first batch of post office code

## References

* #823

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/MessageHub/AvailableChargeConfirmationDataFactoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        public async Task CreateAsync_WhenCalledWithAcceptedEvent_ReturnsAvailableData(
+            [Frozen] Mock<IMessageMetaDataContext> messageMetaDataContext,
+            ChargeCommandAcceptedEvent acceptedEvent,
+            Instant now,
+            AvailableChargeConfirmationDataFactory sut)
+        {
+            // Arrange
+            messageMetaDataContext.Setup(
+                    m => m.RequestDataTime)
+                .Returns(now);
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/MessageHub/AvailableChargeConfirmationDataFactoryTests.cs:44]: Single line, perhaps? :-)

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/MessageHub/AvailableChargeConfirmationDataFactoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+            // Act
+            var actualList =
+                await sut.CreateAsync(acceptedEvent);
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/MessageHub/AvailableChargeConfirmationDataFactoryTests.cs:48]: Single line

**Review by HenrikSommer — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/MessageHub/AvailableChargeConfirmationDataFactoryTests.cs (hunk 1 above)
Comment (bemwen) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/MessageHub/AvailableChargeConfirmationDataFactoryTests.cs]: Changed

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/MessageHub/AvailableChargeConfirmationDataFactoryTests.cs (hunk 2 above)
Comment (bemwen) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/MessageHub/AvailableChargeConfirmationDataFactoryTests.cs]: Changed

**Pull request merged**